migration/migration/012-tracing.sql

```diff
@@ -58,10 +58,10 @@ CREATE TABLE _ps_trace.tag
     tag_type ps_trace.tag_type NOT NULL,
     key_id bigint NOT NULL,
     key ps_trace.tag_k NOT NULL REFERENCES _ps_trace.tag_key (key) ON DELETE CASCADE,
-    value ps_trace.tag_v NOT NULL,
-    UNIQUE (key, value) INCLUDE (id, key_id)
+    value ps_trace.tag_v NOT NULL
 )
 PARTITION BY HASH (key);
+CREATE UNIQUE INDEX tag_key_value_id_key_id_key ON _ps_trace.tag (key, pg_catalog.sha512(value::TEXT::BYTEA)) INCLUDE (id, key_id);
```

Comment:
I believe we can cast jsonb straight to bytea and forgo the cast to text first, right?

Reply:
Perhaps I'm missing something, but I get `ERROR: cannot cast type jsonb to bytea` when I run `select sha512('{"test": 1}'::jsonb::bytea);`

Reply:
But that's actually a very good point: I'm not so sure a textual representation of JSONB is deterministic across major PG versions. Maybe it's worth defining a custom conversion.

Reply:
Turns out `jsonb_send` and `::text` are practically the same https://doxygen.postgresql.org/jsonb_8c_source.html#l00148 And AFAICT there hasn't been a commitment to keep behaviour across pg versions.